### DISCUSSION TITLE
What is Meta DDS?

### DISCUSSION THREAD

**Discussion by Quincunx271** [Quincunx271/meta-dds#9] [Plans & Designs]
[Spawned as an idea from DDS proper](https://github.com/vector-of-bool/dds/discussions/83#discussioncomment-1147075), Meta DDS aims to ease the use of DDS by providing a way to use libraries which are incompatible with DDS. In particular, the idea is to allow libraries with a CMake build system to be imported by DDS as if they supported DDS.

This is accomplished by invoking `cmake` to evaluate all platform-specific / code generation / etc. behavior and create a pure sdist that DDS can consume, albeit a sdist constrained to the particular environment for which CMake was configured. Thus, it is not intended to replace porting a library to DDS, but rather it is intended as a duct tape solution to allow usage of a library while the details of porting to DDS can be figured out.

That's not to say that Meta DDS supports just any CMake project, however. Rather, the project must follow certain guidelines:
 - (to be determined; there are certain expectations in the code, but I can't recall them).

In short, Meta DDS is a step before DDS to produce a build that DDS can consume.

----

One last point: because Meta DDS is effectively a preprocessor for DDS, one thing it will be able to do is run any code generation tools and any other setup needed to convert a project into one that DDS supports. Just how this will be accomplished is currently undecided.

----

An additional feature which is strongly desired: a mechanism to more easily create true DDS sdists from CMake projects. It turns out that most are most of the way there, so it should be possible for a tool to automate this as much as possible.
